src/bosh_azure_cpi/spec/unit/models/disk_id_spec.rb

```diff
@@ -0,0 +1,118 @@
+            expect(disk_id.caching).to eq(caching)
+            expect(disk_id.storage_account_name).to eq(storage_account_name)
+            expect(disk_id.resource_group_name).to eq(default_resource_group_name)
+          end
+        end
+
+        context 'when the id is snapshot id' do
+          let(:id_str) { "bosh-data-#{storage_account_name}-#{uuid}-#{caching}--snapshottime" }
+          it 'should return a correct snapshot id' do
+            snapshot_id = Bosh::AzureCloud::DiskId.parse(id_str, default_resource_group_name)
+            expect(snapshot_id.disk_name).to eq(id_str)
+          end
+        end
+      end
+
+      context 'with managed disks' do
+        let(:id_str) { "bosh-disk-data-#{uuid}-#{caching}" }
+        it 'should return a correct disk id' do
+          disk_id = Bosh::AzureCloud::DiskId.parse(id_str, default_resource_group_name)
+          expect(disk_id.disk_name).to eq(id_str)
+          expect(disk_id.caching).to eq(caching)
+          expect(disk_id.resource_group_name).to eq(default_resource_group_name)
+          expect do
+            disk_id.storage_account_name
+          end.to raise_error /This function should only be called for unmanaged disks/
+        end
+      end
+    end
+
+    context 'when id_str is v2 format' do
+      context 'with unmanaged disks' do
+        let(:storage_account_name) { '6slyzrx7ypji2cfdefaultsa' } # There should be no "-" in the storage account name
+        context 'without resource_group_name' do
+          context 'when the id is disk id' do
+            let(:disk_name) { "bosh-data-#{uuid}" }
+            let(:id_str) { "disk_name:#{disk_name};caching:#{caching};storage_account_name:#{storage_account_name}" }
+            it 'should return a correct disk id' do
+              disk_id = Bosh::AzureCloud::DiskId.parse(id_str, default_resource_group_name)
+              expect(disk_id.disk_name).to eq(disk_name)
+              expect(disk_id.caching).to eq(caching)
+              expect(disk_id.storage_account_name).to eq(storage_account_name)
+              expect(disk_id.resource_group_name).to eq(default_resource_group_name)
+            end
+          end
+
+          context 'when the id is snapshot id' do
+            let(:snapshot_disk_name) { "bosh-data-#{uuid}--snapshottime" }
+            let(:id_str) { "disk_name:#{snapshot_disk_name};caching:#{caching};storage_account_name:#{storage_account_name}" }
+            it 'should return a correct snapshot id' do
+              snapshot_id = Bosh::AzureCloud::DiskId.parse(id_str, default_resource_group_name)
+              expect(snapshot_id.disk_name).to eq(snapshot_disk_name)
+            end
+          end
+        end
+
+        context 'with resource_group_name' do
+          let(:disk_name) { "bosh-data-#{uuid}" }
+          let(:id_str) { "disk_name:#{disk_name};caching:#{caching};resource_group_name:#{default_resource_group_name};storage_account_name:#{storage_account_name}" }
```

Comment:
better not use `default_resource_group_name` in this case, suggest to use another rg name here.

Reply:
As we talked, for the unmanaged disk v2 format, the RG has to be the default resource group name. That's why we didn't add it previously.
That's also why I use `default_resource_group_name` here. Of course, `disk_id.rb` doesn't care about it. So, I'm fine to change it to another rg name.